src/compiler/crystal/semantic/top_level_visitor.cr

```diff
@@ -676,10 +676,10 @@ class Crystal::TopLevelVisitor < Crystal::SemanticVisitor
           if counter == 0 # In case the member is set to 0
             1
           else
-            counter * 2
+            counter &* 2
           end
         else
-          counter + 1
+          counter &+ 1
```

Comment:
An issue to fix or discuss this behaviour should be created.

Reply:
Done at #7268. Is there any other feedback of the changes of the PR?

Reply:
No, the PR is good! Thanks!